+dj/+store_plugins/S3.m

```diff
@@ -0,0 +1,261 @@
+                    'default', [2; 2] ...
+                ), ...
+            'endpoint', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'bucket', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'access_key', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'secret_key', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'secure', struct( ...
+                    'mode', @(datajoint_type) 0, ...
+                    'type_check', @(self) islogical(self), ...
+                    'default', false ...
+                ) ...
+            )
+        backward_validation_config = struct( ...
+            'protocol', struct( ...
+                    'mode', @(unused) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'location', struct( ...
+                    'mode', @(unused) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'subfolding', struct( ...
+                    'mode', @(unused) 0, ...
+                    'type_check', @(self) all(floor(self) == self), ...
+                    'default', [2; 2] ...
+                ), ...
+            'endpoint', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'bucket', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'access_key', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'secret_key', struct( ...
+                    'mode', @(datajoint_type) 1, ...
+                    'type_check', @(self) ischar(self) ...
+                ), ...
+            'secure', struct( ...
+                    'mode', @(datajoint_type) 0, ...
+                    'type_check', @(self) islogical(self), ...
+                    'default', false ...
+                ) ...
+            )
+    end
+    properties
+        protocol
+        datajoint_type
+        location
+        type_config
+        endpoint
+        bucket
+        access_key
+        secret_key
+        web_protocol
+    end
+    methods
+        function self = S3(config)
+            % initialize store
+            self.protocol = config.protocol;
+            [~, start, ~] = regexp(config.location, '[a-zA-Z0-9][a-zA-Z0-9]', 'match', ...
```

Comment:
`fileparts` was considered here but the issue is that it does not form an array of all the directories in the path and there is no OO standard for parsing paths in MATLAB. 

What we are achieving here is identifying where the relevant piece of `location` starts and all that is needed. Since location can be specified both in `unix`-based and `windows`-based paths (and also with a root and without), we cannot reliability look for `/` or `\`. Which means we defer to a 2 character sequence of permissible directory characters.

Reply:
but is this always these two characters? other characters may be used or just one character may be used.

Reply:
Maybe if I describe the intent we can identify if is there is a better approach or not.

As an example, consider the following allowed values for `location`:
`D:\top\sub`
`/top/sub`
`top\sub`
`top/sub`

The objective is to modify this to `top/sub` in either case. Currently, the assumption is that windows drives cannot be more than 1 character and that the `top` directory name has at least 2 characters...

Reply:
What about `/a_/1/` or `/0/3`?

Reply:
Yes, those would be missed. I'll see if I can include a better approach on the next update.